win32ss/user/ntuser/window.h

```diff
@@ -124,4 +124,21 @@ BOOL FASTCALL IntBroadcastImeShowStatusChange(PWND pImeWnd, BOOL bShow);
 VOID FASTCALL IntNotifyImeShowStatus(PWND pImeWnd);
 VOID FASTCALL IntCheckImeShowStatusInThread(PWND pImeWnd);
 
+static inline
+VOID
+WndSetOwner(PWND pwnd, PWND pwndOwner)
```

Comment:
```suggestion
WndSetOwner(_In_ PWND pwnd, _In_opt_ PWND pwndOwner)
```

Reply:
For the record this kind of assignment in windows is called an assignment lock and this scheme is used in a lot of places, not just owner windows.

Reply:
When our win32k is compiled in c++ we can create an object for this.

Reply:
Yes, optimally all window pointers should be `shared_ptr`. But I fear to get there is a long way and probably means rewriting half of the USER code in C++.